gui/widgets/tab.cpp

```diff
@@ -360,23 +384,59 @@ void TabWidget::reflowLayout() {
 
 	int x = _w - _butRP - _butW * 2 - 2;
 	int y = _butTP - _tabHeight;
+
 	_navLeft->resize(x, y, _butW, _butH);
 	_navRight->resize(x + _butW + 2, y, _butW, _butH);
 }
 
 void TabWidget::drawWidget() {
 	Common::Array<Common::String> tabs;
 	Common::Array<int> widths;
-	for (int i = _firstVisibleTab; i <= _lastVisibleTab; ++i) {
-		tabs.push_back(_tabs[i].title);
-		widths.push_back(_tabs[i]._tabWidth);
+	if (g_gui.useRTL()) {
+		for (int i = _firstVisibleTab; i <= _lastVisibleTab; ++i) {
```

Comment:
Please, rewrite it, so you're using only `push_back`. Use trick with `int idx = _lastVisibleTab - i` which I showed you recently. `insert_at(0, ...)` is ugly and expensive.